src/modules/style.ts

```diff
@@ -1,114 +1,215 @@
+    ) {
+      scheduleNextFrame();
+    }
+  });
+
+  if (areObjectsDifferent(delayed, delayedStylesForElement.queued)) {
+    delayedStylesForElement.queued = delayed;
+    scheduleNextFrame();
   }
-  for (name in style) {
-    cur = style[name];
-    if (name === "delayed" && style.delayed) {
-      for (const name2 in style.delayed) {
-        cur = style.delayed[name2];
-        if (!oldHasDel || cur !== (oldStyle.delayed as any)[name2]) {
-          setNextFrame((elm as any).style, name2, cur);
-        }
-      }
-    } else if (name !== "remove" && cur !== oldStyle[name]) {
-      if (name[0] === "-" && name[1] === "-") {
-        (elm as any).style.setProperty(name, cur);
+
+  const stylesToRemove: string[] = [];
+  Object.keys(oldDelayed).forEach((key) => {
+    if (!delayed[key]) {
+      if (!newBaseStyles[key]) {
+        // styles being removed from `delayed` are removed synchronously
+        stylesToRemove.push(key);
       } else {
-        (elm as any).style[name] = cur;
+        // revert back to the base style
+        stylesToAdd[key] = newBaseStyles[key];
       }
     }
-  }
-}
+  });
+  Object.keys(oldBaseStyles).forEach((key) => {
+    if (!newBaseStyles[key] && !delayedStylesForElement.active[key]) {
+      stylesToRemove.push(key);
+    }
+  });
 
-function applyDestroyStyle(vnode: VNode): void {
-  let style: any;
-  let name: string;
-  const elm = vnode.elm;
-  const s = (vnode.data as VNodeData).style;
-  if (!s || !(style = s.destroy)) return;
-  for (name in style) {
-    (elm as any).style[name] = style[name];
-  }
+  Object.keys(stylesToAdd).forEach((key) =>
+    addStyle(elm, key, stylesToAdd[key]!)
+  );
+
+  stylesToRemove.forEach((key) => removeStyle(elm, key));
+};
+
+function onDestroy(vnode: VNode): void {
+  delayedStylesByElement = delayedStylesByElement.filter(
+    ({ elm }) => elm !== vnode.elm
+  );
 }
 
 function applyRemoveStyle(vnode: VNode, rm: () => void): void {
-  const s = (vnode.data as VNodeData).style;
-  if (!s || !s.remove) {
+  const s = vnode.data?.style;
+  if (!s || !s.remove || !(vnode.elm instanceof HTMLElement)) {
     rm();
     return;
   }
   if (!reflowForced) {
     // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-    (vnode.elm as any).offsetLeft;
+    vnode.elm.offsetLeft;
     reflowForced = true;
   }
-  let name: string;
   const elm = vnode.elm;
-  let i = 0;
   const style = s.remove;
   let amount = 0;
   const applied: string[] = [];
-  for (name in style) {
-    applied.push(name);
-    (elm as any).style[name] = style[name];
-  }
-  const compStyle = getComputedStyle(elm as Element);
-  const props = (compStyle as any)["transition-property"].split(", ");
-  for (; i < props.length; ++i) {
+  Object.keys(style).forEach((key) => {
+    applied.push(key);
+    elm.style[key as any] = style[key]!;
+  });
+  const compStyle = getComputedStyle(elm);
+  const props = compStyle.transitionProperty.split(", ");
```

Comment:
this used to be `transition-property`, which I think was a mistake? Typescript thinks `transition-property` doesn't exist anyway and the tests still pass

Reply:
this also goes away with https://github.com/snabbdom/snabbdom/pull/1030